tests/templates/kuttl/smoke/31-assert.yaml

```
@@ -7,6 +7,9 @@ commands:
   # Test envOverrides
   #
   - script: |
+      kubectl -n $NAMESPACE get cm superset-node-default -o yaml | yq -e '.data."superset_config.py"' | grep "COMMON_HEADER_VAR = "group-value""
+      kubectl -n $NAMESPACE get cm superset-node-default -o yaml | yq -e '.data."superset_config.py"' | grep "ROLE_FOOTER_VAR = "role-value""
+      if [ -z "$(kubectl -n $NAMESPACE get cm superset-node-default -o yaml | yq -e '.data."superset_config.py"' | grep "ROLE_HEADER_VAR = \"role-value\"")" ]; then (echo "Success, ROLE_HEADER_VAR not present") else { echo "Failure, ROLE_HEADER_VAR present"; exit 1; } fi
```

I think this should be simplified. Will paste an example in a moment.

---

- Convert to multi-line for readability.
- Refer to `superset_config.py` inside `[]` instead of dotting it.
- Use single quotes so we don't need to escape double quotes.
- Removed the subshell `()` and braces `{}` from the then/else blocks.

```suggestion
      RESULT=$(
          kubectl -n $NAMESPACE get cm superset-node-default -o yaml \
          | yq -e '.data["superset_config.py"]' \
          | grep 'ROLE_HEADER_VAR = "role-value"'
      )
      
      if [ -z "$RESULT" ]; then
        echo "Success, ROLE_HEADER_VAR not present"
      else 
        2>& echo "Failure, ROLE_HEADER_VAR present"
        exit 1
      fi

```

---

I actually think the whole if block could be simplified by using `grep -v` so the test looks more like the others.

---

I will post a separate comment with what I suggest the whole script should look like